hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/HeatMap.java

```diff
@@ -45,10 +45,12 @@ public void createHistogram(final IColumn columnD1, final IColumn columnD2,
         final IRowIterator myIter = membershipSet.getIterator();
         int currRow = myIter.getNextRow();
         while (currRow >= 0) {
-            if ((columnD1.isMissing(currRow)) || (columnD2.isMissing(currRow))){
-                if (!columnD1.isMissing(currRow))  //only column 2 is missing
+            boolean isMissingD1 = columnD1.isMissing(currRow);
+            boolean isMissingD2 = columnD2.isMissing(currRow);
```

Comment:
In general the compiler does not know whether a function has side-effects, so it cannot do common-subexpression elimination for method calls.
But this is not just about efficiency, it is also helping code readability. It's a win-win.